crates/issue/src/lib.rs

```diff
@@ -292,25 +292,48 @@ impl<T: Config> Module<T> {
 
         let mut total_amount = issue.amount + issue.fee;
         ext::btc_relay::verify_transaction_inclusion::<T>(tx_id, merkle_proof)?;
-        let (refund_address, amount_transferred) = ext::btc_relay::validate_transaction::<T>(
-            raw_tx,
-            TryInto::<u64>::try_into(total_amount).map_err(|_e| Error::<T>::TryIntoIntError)?
-                as i64,
-            issue.btc_address,
-            None,
-        )?;
+        let (refund_address, amount_transferred) =
+            ext::btc_relay::validate_transaction::<T>(raw_tx, None, issue.btc_address, None)?;
 
-        if ext::vault_registry::is_vault_liquidated::<T>(&issue.vault)? {
-            // if liquidated, don't try refunds
-            let amount_including_fee = issue
-                .amount
-                .checked_add(&issue.fee)
-                .ok_or(Error::<T>::ArithmeticOverflow)?;
-            ext::vault_registry::issue_tokens::<T>(&issue.vault, amount_including_fee)?;
+        let amount_transferred = Self::u128_to_btc(amount_transferred as u128)?;
+
+        if amount_transferred < total_amount {
+            // only the requester of the issue can execute payments with insufficient amounts
+            ensure!(requester == executor, Error::<T>::InvalidExecutor);
+
+            // decrease the to-be-issued tokens that will not be issued after all
+            let deficit = total_amount
+                .checked_sub(&amount_transferred)
+                .ok_or(Error::<T>::ArithmeticUnderflow)?;
+            ext::vault_registry::decrease_to_be_issued_tokens::<T>(&issue.vault, deficit)?;
+
+            // slash/release griefing collateral proportionally to the amount sent
+            let released_collateral = ext::vault_registry::calculate_collateral::<T>(
+                issue.griefing_collateral,
+                amount_transferred,
+                total_amount,
+            )?;
+            ext::collateral::release_collateral::<T>(&requester, released_collateral)?;
+            let slashed_collateral = issue
+                .griefing_collateral
+                .checked_sub(&released_collateral)
+                .ok_or(Error::<T>::ArithmeticUnderflow)?;
+            ext::vault_registry::slash_collateral::<T>(
+                CurrencySource::Griefing(issue.requester.clone()),
+                CurrencySource::Backing(issue.vault.clone()),
```

Comment:
Do we want to transfer to the fee pool here?

Reply:
Yea I was thinking the same. I kept it like this for consistency the current behavior, but I'll make a new ticket for changing it